index.d.ts

```diff
@@ -10,7 +10,15 @@ Emittery also accepts an array of strings and symbols as event names.
 */
 type EventNames = EventName | readonly EventName[];
 
-declare class Emittery {
+type EventNameFromDataMap<EventDataMap> = Extract<
+	keyof EventDataMap,
+	EventName
+>;
+
+declare class Emittery<
+	EventDataMap extends Emittery.Events = { [key: string]: unknown },
+	EmptyEvents extends EventName = string
```

Comment:
This is a change in the default behaviour. The `Typed` class used `never` as the default. With this change, I think any emittery instance that doesn't explicitly define empty event names will be allowed to emit anything. There should probably be a test for this.

IMO the behaviour should be: allow any string if the non-empty events do. Otherwise default to never. Achievable with some conditional types I think.

Reply:
Yeah, I wasn't exactly sure what the default should be. I went with something permissive to match the way that the node stdlib `EventEmitter` works, and made strict typing opt in by default. My intention with that was to not break importing `Emittery` or using it from JS land with the typescript language service. As you identified it does indeed break `Emittery.Typed`. I think you're right and that I might be able to create a passthrough type that restricts the generics a bit more and matches the old behaviour as well.

Reply:
This is probably good enough: 
```suggestion
	EmptyEvents extends EventName = string extends keyof EventDataMap ? string : never
```

When EventDataMap is the default of `Record<string, unknown>`, you get `string`. If it's not the default (i.e. a user-specified map), you get `never`.